src/consensus/consensus.h

```diff
@@ -12,7 +12,13 @@
 static const unsigned int MAX_BLOCK_SERIALIZED_SIZE = 4000000;
 /** The maximum allowed weight for a block, see BIP 141 (network rule) */
 static const unsigned int MAX_BLOCK_WEIGHT = 4000000;
-/** The maximum allowed size for a block excluding witness data, in bytes (network rule) */
+/**
+ * The maximum allowed size for a block excluding witness data, in bytes (network rule).
+ * This parameter is largely superfluous because it is directly implied by the above block
+ *  weight limit, even when BIP 141 is not active.  It continues to exist for use in
+ *  various early tests that run before the witness data has been checked.
```

Comment:
"tests" is overloaded and could be confused for unit/integration tests. "early checks" might be better.

Why are those checks not removed if they don't change the rules? Would be good to mention the reason for it also here.